🏗 Run local integration tests on minified builds by default

rsimha opened the pull request:
This PR does the following:
- Performs `gulp dist --fortesting` as a prerequisite task for `gulp test --integration` (when `--nobuild` is not passed in)
- Performs `gulp clean` before running `gulp build`, `gulp dist`, and `gulp check-types`
- Uses `del` instead of `rimraf` since it's [newer and better](https://www.npmjs.com/package/del)
- Removes a couple of unused `gulp-help` instances

Future work: Run Visual tests (#22391) and E2E tests (#22392) against minified builds.

Follow up to #22348

rsimha commented:
Bumping this.

choumx commented:
Any reason why we added `clean()` to the "build" task? 

It's breaking some workflows for testing shadow-v0.js (amp-shadow.js) since `gulp --extensions=amp-foo` doesn't build it. `gulp build` does, but it will get cleaned by the former.

rsimha commented:
> Any reason why we added `clean()` to the "build" task?
> 
> It's breaking some workflows for testing shadow-v0.js (amp-shadow.js) since `gulp --extensions=amp-foo` doesn't build it. `gulp build` does, but it will get cleaned by the former.

I added `clean()` to `dist()` and `checkTypes()` because they're now run by default during local PR checks, and I thought it made sense to start running tests from a clean state. Adding `clean()` to `build()` was an after-thought for consistency.

Happy to remove `clean()` from `build()`, or if you prefer, make `build` also build the shadow runtime by default. In fact, this is something we discussed just today in https://github.com/ampproject/amphtml/pull/22397#discussion_r286617921
